src/main/kotlin/listener/StarboardListener.kt

```diff
@@ -0,0 +1,127 @@
+package tech.trip_kun.sinon.listener
+
+import com.j256.ormlite.dao.Dao
+import net.dv8tion.jda.api.EmbedBuilder
+import net.dv8tion.jda.api.JDA
+import net.dv8tion.jda.api.entities.Member
+import net.dv8tion.jda.api.entities.Message
+import net.dv8tion.jda.api.entities.emoji.Emoji
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent
+import net.dv8tion.jda.api.events.message.react.*
+import net.dv8tion.jda.api.exceptions.ErrorResponseException
+import net.dv8tion.jda.api.hooks.ListenerAdapter
+import net.dv8tion.jda.api.requests.GatewayIntent
+import net.dv8tion.jda.api.requests.GatewayIntent.*
+import org.apache.commons.logging.Log
+import tech.trip_kun.sinon.Logger
+import tech.trip_kun.sinon.annotations.ListenerClass
+import tech.trip_kun.sinon.annotations.ListenerConstructor
+import tech.trip_kun.sinon.annotations.ListenerIntents
+import tech.trip_kun.sinon.data.entity.Guild
+import tech.trip_kun.sinon.data.entity.StarboardEntry
+import tech.trip_kun.sinon.data.getGuildDao
+import tech.trip_kun.sinon.data.getStarboardEntryDao
+import tech.trip_kun.sinon.data.runSQLUntilMaxTries
+@ListenerClass
+@ListenerIntents(GatewayIntent.GUILD_MESSAGE_REACTIONS)
+@ListenerIntents(MESSAGE_CONTENT)
+class StarboardListener @ListenerConstructor constructor(private val jda: JDA): ListenerAdapter() {
+    init {
+        Logger.info("StarboardListener initialized")
+    }
+    override fun onMessageReactionRemoveEmoji(event: MessageReactionRemoveEmojiEvent) {
+        if (!event.isFromGuild) return
+        val guildChannel  = event.guildChannel
+        val message = guildChannel.retrieveMessageById(event.messageId).complete()
+        commonWork(event.guild, message)
```

Comment:
I think what I am going to do is to try to find a library (possibly jda-ktx) that integrates coroutines/suspend functions into the JDA ecosystem. This should help avoid callback hell, which is something I don't want to engage in, especially because .complete() cannot be used in .queue()